### PR TITLE
historical feature update

### PR DESCRIPTION
**Pull request by wintonzheng**
- [ ] Does this PR have impact on local development experience? If yes, make sure you have a plan and add the documentations to address issues that come with the change
- [ ] bump version
- [ ] make a release
- [ ] publish to pypi service

### CODE REVIEW

File: wyvern/feature_store/feature_server.py [hunk 1]
```diff
@@ -456,6 +460,140 @@ async def get_historical_features(
             results=final_df.to_dict(orient="records"),
         )
 
+    @app.post(f"{settings.WYVERN_HISTORICAL_FEATURES_PATH}_v2")
```
Comment (suchintan) [inline wyvern/feature_store/feature_server.py:465]: Should the version here follow the standard `api/v2/get_historical_features`

File: wyvern/feature_store/feature_server.py [hunk 2]
```diff
@@ -456,6 +460,140 @@ async def get_historical_features(
...
+        #     )
+        # for key, value in data.entities.items():
+        #     if len(value) != length_of_requests:
+        #         raise HTTPException(
+        #             status_code=400,
+        #             detail=f"Length of requests({length_of_requests}) and {key}({len(value)}) should be the same",
+        #         )
+
+        # Generate a 10-digit hex for the request
+        hex_id = secrets.token_hex(5)
```
Comment (suchintan) [inline wyvern/feature_store/feature_server.py]: What's a hex?

File: wyvern/feature_store/feature_server.py (hunk 1 above)
Comment (wintonzheng) [reply wyvern/feature_store/feature_server.py:465]: yes we can do `f"v2/{settings.WYVERN_HISTORICAL_FEATURES_PATH}"`

Also:
this API path is just a placeholder. eventually it should replace the original API.

And this service is the feature store service, not the pipeline service. so it doesn't follow the `api/vX/` pattern

File: wyvern/feature_store/feature_server.py (hunk 2 above)
Comment (wintonzheng) [reply wyvern/feature_store/feature_server.py]: it's just a random hexadecimal number, which is part of the temporary/transient table name we're generating

File: wyvern/feature_store/feature_server.py [hunk 3]
```diff
@@ -456,6 +460,140 @@ async def get_historical_features(
...
+                logger.warning(f"feature={realtime_feature} is not found")
+                continue
+
+            if len(entity_names) == 2:
+                entity_name_1 = entity_names[0]
+                entity_name_2 = entity_names[1]
+
+                # TODO: valid entity_name_1 and entity_name_2 are in the table columns
+                # if entity_name_1 not in data.entities:
+                #     logger.warning(
```
Comment (ykeremy) [inline wyvern/feature_store/feature_server.py]: I think these should be 400 Bad Request responses

File: wyvern/feature_store/feature_server.py [hunk 4]
```diff
@@ -456,6 +462,62 @@ async def get_historical_features(
...
+        )
+        composite_historical_feature_table = f"HISTORICAL_FEATURES_{random_id}"
+
+        select_sql = f"""
+        CREATE TEMPORARY TABLE {composite_historical_feature_table} AS
+        SELECT *, {composite_columns}, TIMESTAMP as event_timestamp
+        FROM {data.table}
+        """
+        snowflake_ctx = generate_snowflake_ctx()
+        snowflake_ctx.cursor().execute(select_sql)
```
Comment (ykeremy) [inline wyvern/feature_store/feature_server.py:503]: should we handle some errors and add retry logic?

File: wyvern/feature_store/historical_feature_util.py [hunk 5]
```diff
@@ -117,6 +118,100 @@ def build_historical_real_time_feature_requests(
     return result_dict
 
 
+def build_and_merge_realtime_pivot_tables(
```
Comment (ykeremy) [inline wyvern/feature_store/historical_feature_util.py:121]: we should create a one pager that explains the logic clearly with examples. we can even get sth together using the stuff we went over to clarify the logic

Comment (suchintan) [reply wyvern/feature_store/historical_feature_util.py:121]: https://docs.wyvern.ai/training_data

Maybe here?

File: wyvern/feature_store/feature_server.py (hunk 4 above)
Comment (wintonzheng) [reply wyvern/feature_store/feature_server.py:503]: I'm not sure when we should retry tho. Let's start with busting the API if there's any error. most likely error comes from bad data in the input table (entity table), which probably wont' be fixed by retry. 

let's launch this version without error handling and start catching issues with matias' data.

File: wyvern/feature_store/historical_feature_util.py (hunk 5 above)
Comment (wintonzheng) [reply wyvern/feature_store/historical_feature_util.py:121]: good point. i'll write something after i test this endpoint

**Pull request closed without merging**